### PR TITLE
fix: make register buttons corners rounded

### LINKED ISSUE #1020: Change rectangular button to have rounded corners
**Describe the chore**

The button for tickets in the EventDetailsFragment has rounded corners:

<table>
        <tr>
<td></td>
        </tr>
</table>

Similar buttons to register for the event in the following fragments lack rounded corners
<table>
        <tr>
<td></td>
<td></td>
        </tr>
</table>

I would like to make the corners for these buttons rounded as well for consistency.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by addiegupta**
Fixes #1020 

Changes: 
- Modified "register" buttons in fragment_attendee and fragment_tickets to Material Buttons 
and then applying rounded corners.
- Changed CardView of Tickets button in fragment_event to MaterialButton

Screenshots for the change:

ezgif com-video-to-gif

### CODE REVIEW

**Comment by addiegupta** [comment]
@nikit19 what changes should I make? The tickets button on the previous fragment looks the same,so I made these to look consistent

**Comment by addiegupta** [comment]
> The button doesn't look good in my opinion

@nikit19 what changes should I make? The tickets button on the previous fragment looks the same,so I made these to look consistent

**Comment by nikit19** [comment]
@addiegupta try adjusting the width. I think it is a little different. If not then it is fine

**Comment by addiegupta** [comment]
> @addiegupta try adjusting the width. I think it is a little different. If not then it is fine

The layout_width is set to "match_parent" for all and  layout_margin is 16dp for all

**Comment by addiegupta** [comment]
I have made the required changes. Please review @nikit19

**Comment by addiegupta** [comment]
Do i need to make any changes @nikit19  @iamareebjamal ?

**Comment by liveHarshit** [comment]
resolve conflicts

**Comment by addiegupta** [comment]
@liveHarshit done

**Comment by addiegupta** [comment]
@liveHarshit as app theme has been changed to material components in #1255 , can this be merged now or should I make any changes?

**Comment by liveHarshit** [comment]
Okay, now you can use material button instead.

**Comment by liveHarshit** [comment]
https://material.io/design/components/buttons.html#hierarchy-placement
https://material.io/develop/android/components/material-button/

**Comment by addiegupta** [comment]
@liveHarshit used MaterialButton and updated GIF

**Comment by iamareebjamal** [comment]
Buttons should have been blue. Please correct app theme to utilize secondary color for buttons

**Comment by addiegupta** [comment]
> Buttons should have been blue. Please correct app theme to utilize secondary color for buttons

According to the Material Components Guidelines "The baseline color for contained, text and outlined buttons is your primary color." Should I individually set the colors of the buttons to blue?

**Comment by iamareebjamal** [comment]
You should set it in the primary style

**Comment by addiegupta** [comment]
> You should set it in the primary style

In the latest commit , I have created a new style, setting `backgroundTint` as the secondary color and then used this style as the `materialButtonStyle` in the primary `AppTheme` style. Setting `colorButtonNormal` directly in the primary `AppTheme` did not apply the color to the new MaterialButtons

**Comment by addiegupta** [comment]
> You should set it in the primary style

In the latest commit , I have created a new style, setting `backgroundTint` as the secondary color and then used this style as the `materialButtonStyle` in the primary `AppTheme` style. Setting `colorButtonNormal` directly in the primary `AppTheme` did not apply the color to the new MaterialButtons

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
+        <item name="materialButtonStyle">@style/AppTheme.MaterialButton</item>
```
Comment (iamareebjamal) [inline app/src/main/res/values/styles.xml]: Move this down below colors

Comment (addiegupta) [reply app/src/main/res/values/styles.xml]: okay sure

**Comment by addiegupta** [comment]
@iamareebjamal changes made and latest upstream merged

**Comment by aman210697** [comment]
Move Textcolor, textstyle, cornerradius, textsize, elevation to button style.

**Comment by iamareebjamal** [comment]
Create a new style for this

**Comment by addiegupta** [comment]
> Create a new style for this

A new style in addition to the one that I have created for `backgroundTint`?

**Comment by iamareebjamal** [comment]
Yes

**Comment by addiegupta** [comment]
@iamareebjamal done

**Pull request merged**